test/e2e/cloud-provider-azure.go

```diff
@@ -45,7 +45,13 @@ func InstallCalicoAndCloudProviderAzureHelmChart(ctx context.Context, input clus
 	specName := "cloud-provider-azure-install"
 	By("Installing the correct version of cloud-provider-azure components via helm")
 	options := &helmVals.Options{
-		Values: []string{fmt.Sprintf("infra.clusterName=%s", input.ConfigCluster.ClusterName), fmt.Sprintf("cloudControllerManager.clusterCIDR=%s", strings.Join(cidrBlocks, `,`))},
+		Values: []string{
+			fmt.Sprintf("infra.clusterName=%s", input.ConfigCluster.ClusterName),
+			fmt.Sprintf("cloudControllerManager.clusterCIDR=%s", strings.Join(cidrBlocks, `,`)),
+			// TODO: remove extra config once the azure-cloud-provider Helm chart release supports VMSS Flex.
+			"cloudControllerManager.imageTag=v1.26.0",
```

Comment:
same comment as Tilt here, we should be able to fix this now that cloud provider 1.26 is out, but we shouldn't use v1.26 unless k8s version is also 1.26 as it may lead to incompatibility. This also means we can't test Flex unless we use k8s v1.26 in theory.

Reply:
In theory, although in practice it seems to work fine with k8s 1.24 and 1.25. Previously we had it hard-coded to a pre-release SHA.

But I agree, we should limit this to k8s >= 1.26. I'll make that change.

Reply:
Could we also document this limitation and the requirement for OOT cloud-provider?